odoo/addons/test_menus/static/src/js/test_menus.js

```diff
@@ -0,0 +1,259 @@
+(function (exports) {
+    /**
+     * The purpose of this test is to click on every installed App and then
+     * open each view.
+     */
+    "use strict";
+    var clientActionCount = 0;
+    var viewUpdateCount = 0;
+    var testedApps;
+    var testedMenus;
+
+    function prepareMonkeys() {
+        var AbstractController = odoo.__DEBUG__.services['web.AbstractController'];
+        var Discuss = odoo.__DEBUG__.services['mail.Discuss'];
+        var WebClient = odoo.__DEBUG__.services["web.WebClient"];
+
+        // Monkey patch the current_action_updated 
+        WebClient.include({ 
+            current_action_updated : function (action, controller) {
+                this._super(action, controller);
+                clientActionCount++;
+            },
+        });
+
+        AbstractController.include({
+            update: function(params, options) {
+                return this._super(params, options).then(function (){
+                    viewUpdateCount++;
+                });
+            },
+        });
+
+        Discuss.include({
+            _fetchAndRenderThread: function() {
+                return this._super().then(function (){
+                    viewUpdateCount++;
+                });
+            },
+        });
+    }
+
+    // Main function that starts orchestration of tests
+    function clickEverywhere(){
```

Comment:
docstring:
```

    /**
     * Test Odoo by clicking on each app, each menu item, each view and each
     * search filters.
     *
     * This test will fail if something crashes.  In that case, it will simply log error in
     * console. Note that this test takes a long time to run, since it is quite extensive.
     *
     * @returns {Deferred} resolved if test is successful, rejected otherwise
     */

```